lib/mongoose_push/service/apns/supervisor.ex

```diff
@@ -0,0 +1,70 @@
+defmodule MongoosePush.Service.APNS.Supervisor do
+  @moduledoc """
+  APNS module supervising Sparrow's PoolSupervisor and APNS State
+  """
+  use Supervisor, id: :apns_supervisor
+  require Logger
+
+  @spec start_link([Keyword.t()]) :: Supervisor.on_start()
```

Comment:
I feel that we need to create a type that describes in-depth how this configuration looks like (`pools_config`). Previously it was used only within one file only, now it's being passed around through multiple layers, one of which is a generic behavior.